Hive threat - new event for xenos

Lumipharon opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
New random event like intel computers, but for xenos.

The Queen Mother determines that an individual human that's deployed groundside is a particular threat to the greater hive. The hive must psydrain (not cocoon, since we're sucking their brain or whatever) them to steal their secrets and counter that current or future threat.

The target has a hud mark to make it easier to keep track of them.
![image](https://user-images.githubusercontent.com/7869430/198007970-102ce243-80da-4231-a0ae-d13206020d84.png)

If successful, the hive as a whole is rewarded with a slight movement speed buff for 2 minutes, triple psypoints for the drain itself, and a full plasma regen and salve heal for every xeno in the hive.
Additionally, the specific xeno that drained the target is further rewarded, being boosted to ancient maturity and their next evo tier.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
More events cool.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Adds a new random event for xenos - Hive threat
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Losenis commented:
Me when it rolls Frank
![image](https://user-images.githubusercontent.com/72712909/197424257-4f7bba00-b79c-4ed7-a51c-8584109f4c46.png)

novaepee commented:
Would be interesting if xenomorphs or admins can select a target. Truly metatarget meme potential.

RipGrayson commented:
> Would be interesting if xenomorphs or admins can select a target. Truly metatarget meme potential.

Make it a smite, should be simple enough to implement once this PR is in.

Derrick2Electricboogaloo commented:
Oh great now I will be really glowing in the dark.

Lumipharon commented:
> Can it notify the target to? Seeing their reaction would be priceless once they know the entire hive wants to strangle em.

I kinda want them to not know, considering they can just play super safe if they do which is boring. Also fluff wise it probs doesn't make much sense.

bowlofcereal commented:
meta targeting mechanics, lol based

Jalopkomit18 commented:
> > Can it notify the target to? Seeing their reaction would be priceless once they know the entire hive wants to strangle em.
> 
> I kinda want them to not know, considering they can just play super safe if they do which is boring. Also fluff wise it probs doesn't make much sense.

I think that the target should be notified. Xenos aren't losing anything if the target is killed. So if they kill the target good, if not no big deal. I doubt the target would sit in fob upon notification, but if they do nothing of value lost.

Jalopkomit18 commented:
Also I think that the rewards might be a little much. When you say that the xeno that drains get upgraded to next evo tier, do you just mean that the evo progress is set to full? Because if they are forcibly evo'd that wouldn't be good

Lumipharon commented:
> Also I think that the rewards might be a little much. When you say that the xeno that drains get upgraded to next evo tier, do you just mean that the evo progress is set to full? Because if they are forcibly evo'd that wouldn't be good

Just fills the bar, yeah. The rewards are pretty arbitrary, I'm sure someone else could make something more interesting.

I don't think benos get notified of intel events either afaik.

FosterGrime-Heymynameslouie commented:
this is insanely cool and would give xenos alot more stuff to play/do with with a dedicated target

any plans on doing a marine version though? maybe double credit sell points on dead beano?

spedde metghe

TiviPlus commented on code/modules/events/hive_threat.dm:
```diff
@@ -0,0 +1,74 @@
+/datum/round_event/hive_threat
+	///The human target for this event
+	var/mob/living/carbon/human/hive_target
+	///The announcement sound
+	var/sound/queen_sound
```
why is this a var on the actual event instead of in the one proc its used?

Lumipharon replied:
I remember doing it for a reason. I uh, can't remember the reason. Originally I may have been making this as a more generic parent event I think.

TiviPlus commented on code/modules/events/hive_threat.dm:
```diff
@@ -0,0 +1,74 @@
+
+///Actually applies the buff to the hive
+/datum/round_event/hive_threat/proc/bless_hive(mob/living/carbon/xenomorph/drainer)
+	for(var/mob/living/carbon/xenomorph/receiving_xeno in GLOB.alive_xeno_list)
```
no as?

TiviPlus commented on code/modules/events/hive_threat.dm:
```diff
@@ -0,0 +1,74 @@
+	//probs add some xenohud overlay to the target so we can keep track of the nerd
+	RegisterSignal(SSdcs, COMSIG_GLOB_HIVE_TARGET_DRAINED, .proc/handle_reward)
+	xeno_message("The Queen Mother senses that [hive_target] is a deadly threat to the hive. Psydrain them for the Queen Mother's blessing!", force = TRUE)
+	for(var/mob/living/carbon/xenomorph/receiving_xeno in GLOB.alive_xeno_list)
```
no as?
also doesnt send sound work with lists anyway

Lumipharon replied:
oh does it? honk.

TiviPlus reviewed (changes requested):
rest lgtm

comfyorange commented:
#### Test Merge Deployed

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 1a7897fb75c0bc591fc470eeb19602cedd44f913
Pull Request: 720849c7551b6db9cb97c49d1bc6195ead2899a6
Server: 39862f7d553c85a605709fa41bd5380a2f39c840

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

comfyorange commented:
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 48bb57ab23241cd1fe25e1c80ec1cfa43da71592
Pull Request: 720849c7551b6db9cb97c49d1bc6195ead2899a6
Server: 9eda513e79e9a6a25c9ef0ba9b659d01bc276131

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

comfyorange commented:
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Lewdcifer

##### Revision
Origin: 48bb57ab23241cd1fe25e1c80ec1cfa43da71592
Pull Request: 720849c7551b6db9cb97c49d1bc6195ead2899a6
Server: e5943caf779b3b116338f723ad981803b280edd2

##### Comment

[GitHub Deployments](https://github.com/tgstation/TerraGov-Marine-Corps/deployments/activity_log?environment=TGS%3A%20TGMC)

breadhunt commented:
PVT John McShitface joining the game and getting marked for fucking death immediately

![image](https://user-images.githubusercontent.com/80229419/199300380-1f63de57-5a3e-4665-b343-63ddaa86ee30.png)
 
> I don't think benos get notified of intel events either afaik.

They do. This should probably be a team-wide announcement considering how significant the rewards could be.

Memes aside, this could pretty easily be given to someone new or incompetent (because funny RNG), in which case, shiiiiit. New players now actively detriment their team more. Time to BE anyone under 50 hours

Losenis commented:
Sacrifice every PVT to the req pit

comfyorange commented:
#### Test Merge Updated

##### Server Instance
TGMC

##### Merged By
Tiviplus

##### Revision
Origin: 48bb57ab23241cd1fe25e1c80ec1cfa43da71592
Pull Request: 720849c7551b6db9cb97c49d1bc6195ead2899a6
Server: fa36ab454b4d3585ace1d08747ae259292dba298

##### Comment

Lumipharon commented:
Removed the limit of one at a time because why not.
Reduced the event weight to 10 and increased min start time to 30 (it can't trigger without humans on the ground anyway).
Blacklisted from crash where you can't drain.
Fixed maturity check so it won't accidentally demature primos.

Lumipharon commented:
> Might want to blacklist synths since they can't get drained, it's a waste of a target.
> 
> ![image](https://user-images.githubusercontent.com/72712909/199633101-5cb946a3-ce9d-467f-87ed-92afbf5c93fa.png)

Curse everything being a child of human.
